site/how-it-works.md

```diff
@@ -95,6 +105,108 @@ the MAC. This way weave peers never need to exchange the MAC addresses
 of clients and need not take any special action for ARP traffic and
 MAC discovery.
 
+### <a name="topology"></a>Topology Communication
+
+Topology is communicated over the TCP links between peers.
+There are two message types:
+
+##### FetchAll
+This carries no payload. The receiver responds with the entire
+topology model as the receiver has it. FetchAll is sent:
+
+ * when a new connection is established, by both sides of the new connection
+ * when an update is received that references a peer that the receiver does not know
+ * periodically, on a timer, in case any updates have been missed
+
+##### Update
+This carries a topology payload. This is sent:
+
+  * upon receipt of a FetchAll message, as above,
+  * when a connection is added - the update will contain the two peers
```

Comment:
We need to be precise in terminology here. 'Update' is sent when the connection is marked as 'established', not when it is added.